Feature/166002062 add question mark tooltip

lingyun1010 opened the pull request:
I add a tooltip based on Pablo's information text and correct the experiment description link in the table content. Please have a review. Thanks.

lingyun1010 committed:
Install react-tooltip pacakge, add an TooltipIcon

lingyun1010 committed:
Add tooltip to download and correct link scope

lingyun1010 committed:
Clean up code

monicajianu commented on src/TableContent.js:
```diff
@@ -2,8 +2,17 @@ import React from 'react'
 import tableHeaderCells from './tableHeaderCells'
+import TooltipIcon from './TooltipIcon'
+
+const downloadTooltip = `<ul>
```
I can't remember the discussion in the planning meeting, but did we agree we'll pass these in as props? This tooltip won't be exactly the same for bulk Atlas.

lingyun1010 replied:
I do not think bulk Atlas is going to use download function yet. The question mark is a temporal solution as we discussed, but I agree a props is always good.

monicajianu commented on src/TableContent.js:
```diff
@@ -24,16 +33,19 @@ const TableContent = ({tableHeader, searchedColumnIndex, searchQuery, orderedCol
-                    Download {checkedRows.length} {checkedRows.length === 1 ? `entry` : `entries`}
-                  </a> :
-                  `Download`
+                  <div>
```
I think the conditional rendering could be improved here, so you don't end up with duplication.

monicajianu reviewed (changes requested):
It does the job, but it would be better if the tooltip text was passed in as a prop. We want to keep the table generic.

lingyun1010 committed:
Set downloadTooltip as props and refactor codes

lingyun1010 committed:
Update tests with downloadTooltip props

alfonsomunozpomer commented on html/index.html:
```diff
@@ -27,6 +27,12 @@
+          <li>  Normalised filtered count matrix after quantification</li>
+          <li>  TPM (transcripts per million RNA molecules - only available for SmartSeq)</li>
+          <li>  SDRF file with the experimental design metadata</li>
+        </ul>`,
```
Be consistent using spaces, in this kind of markup it’s better not to have any between the `<li>`  and the text.

alfonsomunozpomer commented on __test__/ExperimentTable.test.js:
```diff
@@ -16,7 +16,8 @@ describe(`ExperimentTable`, () => {
     enableIndex: true,
-    TableCellDiv: TableCellDiv
+    TableCellDiv: TableCellDiv,
+    downloadTooltip: `tooth`
```
There are two problems here:
1. The test doesn’t convey the fact that the tooltip content can be HTML (and is rendered as such).
2. Just the word `tooth` is too cryptic. If you fell like a dental comment is in order, be a bit more explicit (e.g. `I can’t think of a good tooltip text here, but I can say that my <b>tooth hurts a lot</b>!). Be conscious of future Atlas developers who might not be able to guess why certain hard-coded values were picked!

alfonsomunozpomer reviewed (changes requested):
Two very minor style suggestions. However, coverage has dropped after this PR. Could you fix that, please?

lingyun1010 committed:
Remove spaces between <li> tag and text

lingyun1010 committed:
Clean posttest to show coverage table after test

lingyun1010 committed:
Clean up codes

lingyun1010 committed:
Update download tooltip test and improve coverage

by adding simulation tests for entries of page selection box and search all columns box

alfonsomunozpomer reviewed (approved):
All good! The new tests are a great addition! Revert the `postTest` phase in `package.json` and squas-merge.

lingyun1010 committed:
Revert "Clean posttest to show coverage table after test"

This reverts commit 4e9f163fd110fc5468234bcf1769e136f4e85f76.
